pkg/types/validation/installconfig_test.go

```diff
@@ -0,0 +1,299 @@
+			ClusterNetworks: []netopv1.ClusterNetwork{
+				{
+					CIDR:             "192.168.1.0/24",
+					HostSubnetLength: 4,
+				},
+			},
+		},
+		Machines: []types.MachinePool{
+			{
+				Name: "master",
+			},
+			{
+				Name: "worker",
+			},
+		},
+		Platform: types.Platform{
+			AWS: &aws.Platform{
+				Region: "us-east-1",
+			},
+		},
+		PullSecret: `{"auths":{"example.com":{"auth":"authorization value"}}}`,
+	}
+}
+
+func TestValidateInstallConfig(t *testing.T) {
+	cases := []struct {
+		name          string
+		installConfig *types.InstallConfig
+		valid         bool
+	}{
+		{
+			name:          "minimal",
+			installConfig: validInstallConfig(),
+			valid:         true,
+		},
+		{
+			name: "missing name",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.ObjectMeta.Name = ""
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing cluster ID",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.ClusterID = ""
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid ssh key",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.SSHKey = "bad-ssh-key"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid base domain",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.BaseDomain = ".bad-domain."
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid network type",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.Type = "bad-type"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid service cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ServiceCIDR = ipnet.IPNet{}
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid cluster network cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "bad-cidr"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "overlapping cluster network cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "10.0.0.0/24"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "cluster network host subnet length too large",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "192.168.1.0/24"
+				c.Networking.ClusterNetworks[0].HostSubnetLength = 9
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing master machine pool",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Machines = []types.MachinePool{
+					{
+						Name: "worker",
+					},
+				}
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing worker machine pool",
```

Comment:
Should this be allowed?  We can create worker pools as a day-2 operation, can't we?

Reply:
Fo now our install fails if we do not have workers. So this is required for now.

Reply:
I remember it failing at one point without workers defined in the install config. That may have been resolved since I originally wrote it. I'll dig into it further, and if we succeed without workers defined, then I will file a follow-up PR.